packages/admin/admin/src/common/buttons/save/SaveButton.tsx

```diff
@@ -27,29 +78,40 @@ export interface SaveButtonProps extends ButtonProps {
 
 export type SaveButtonDisplayState = "idle" | "saving" | "success" | "error" | "conflict";
 
-const SaveBtn = ({
-    saving = false,
-    hasErrors = false,
-    hasConflict = false,
-    children = <FormattedMessage {...messages.save} />,
-    savingItem = <FormattedMessage id="comet.saveButton.savingItem.title" defaultMessage="Saving" />,
-    successItem = <FormattedMessage id="comet.saveButton.successItem.title" defaultMessage="Successfully Saved" />,
-    errorItem = <FormattedMessage id="comet.saveButton.errorItem.title" defaultMessage="Save Error" />,
-    conflictItem = <FormattedMessage {...messages.saveConflict} />,
-    saveIcon = <Save />,
-    savingIcon = <ThreeDotSaving />,
-    successIcon = <Check />,
-    errorIcon = <Error />,
-    conflictIcon = <ErrorIcon />,
-    variant = "contained",
-    color = "primary",
-    classes,
-    disabled,
-    ...restProps
-}: SaveButtonProps & WithStyles<typeof styles>) => {
+export function SaveButton(inProps: SaveButtonProps) {
+    const {
+        saving = false,
+        hasErrors = false,
+        hasConflict = false,
+        children = <FormattedMessage {...messages.save} />,
+        savingItem = <FormattedMessage id="comet.saveButton.savingItem.title" defaultMessage="Saving" />,
+        successItem = <FormattedMessage id="comet.saveButton.successItem.title" defaultMessage="Successfully Saved" />,
+        errorItem = <FormattedMessage id="comet.saveButton.errorItem.title" defaultMessage="Save Error" />,
+        conflictItem = <FormattedMessage {...messages.saveConflict} />,
+        saveIcon = <Save />,
+        savingIcon = <ThreeDotSaving />,
+        successIcon = <Check />,
+        errorIcon = <Error />,
+        conflictIcon = <ErrorIcon />,
+        variant = "contained",
+        color = "primary",
+        disabled,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminSaveButton",
+    });
+
     const [displayState, setDisplayState] = React.useState<SaveButtonDisplayState>("idle");
     const saveSplitButton = useSplitButtonContext();
 
+    const ownerState: OwnerState = {
+        savingState: displayState === "saving",
+        errorsState: displayState === "error",
+        successState: displayState === "success",
+        conflictState: displayState === "conflict",
```

Comment:
I'm not fond of this naming. Also, wouldn't it be easier to pass displayState to ownerState instead of introducing four new states?